### PR TITLE
Display parent channel details

### PR DESCRIPTION
**Pull request by Royosef**
Add parent channel info to ChannelExtractor, StreamExtractor: name, url & avatar url

- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I have tested the API against [NewPipe](https://github.com/TeamNewPipe/NewPipe).
- [x] I agree to create a pull request for [NewPipe](https://github.com/TeamNewPipe/NewPipe) as soon as possible to make it compatible with the changed API.

### CODE REVIEW

File: extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamExtractor.java [hunk 1]
```diff
@@ -178,6 +178,39 @@ public StreamExtractor(StreamingService service, LinkHandler linkHandler) {
     @Nonnull
     public abstract String getUploaderAvatarUrl() throws ParsingException;
 
+    /**
+     * The Url to the page of the parent chanel of the stream. This must not be a homepage,
+     * but the page offered by the service the extractor handles. This url will be handled by the
+     * <a href="https://teamnewpipe.github.io/documentation/03_Implement_a_service/#channel">ChannelExtractor</a>,
```
Comment (wb9688) [inline extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamExtractor.java:184]: Use `{@link ChannelExtractor}` instead

File: extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamExtractor.java [hunk 2]
```diff
@@ -178,6 +178,39 @@ public StreamExtractor(StreamingService service, LinkHandler linkHandler) {
...
+     * so be sure to implement that one before you return a value here, otherwise NewPipe will crash if one selects
+     * this url.
+     *
+     * @return the url to the page of the parent chanel of the stream or an empty String
+     * @throws ParsingException
+     */
+    @Nonnull
+    public abstract String getParentChannelUrl() throws ParsingException;
```
Comment (wb9688) [inline extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamExtractor.java:192]: Should we add a `getParentChannelUrl` or a `getSubchannelUrl` here? Same thing for name and avatar URL obviously,

Imho we should add `getSubchannelUrl`, as currently we return the parent channel's URL, so the values returned by `getChannelUrl` stays the same. I want someone else's opinion here first though.

Edit: it looks like you're returning the subchannel's URL here, so that's obviously not correct anyway.

Comment (Royosef) [reply extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamExtractor.java:192]: I'm not really sure myself what will be the right naming. In Peertube it's "account page" and "channel page", maybe we should do the same?
image

File: extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamExtractor.java (hunk 1 above)
Comment (Royosef) [reply extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamExtractor.java:184]: hmm, truth is I just copied the correspond comment from `getUploaderUrl`. How can I use `{@link ChannelExtractor}`?

File: extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamExtractor.java (hunk 2 above)
Comment (wb9688) [reply extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamExtractor.java:192]: @Royosef: No, just change it to `getSubchannelUrl()`. We aren't going to call it accounts and channels as that's unclear and only PeerTube calls it that.

File: extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamExtractor.java (hunk 1 above)
Comment (wb9688) [reply extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamExtractor.java:184]: @Royosef: Just change the whole `<a href="https://teamnewpipe.github.io/documentation/03_Implement_a_service/#channel">ChannelExtractor</a>` to `{@link ChannelExtractor}`.

File: extractor/build.gradle [hunk 3]
```diff
@@ -1,3 +1,6 @@
+compileJava.options.encoding = "UTF-8"
+compileTestJava.options.encoding = "UTF-8"
+
```
Comment (wb9688) [inline extractor/build.gradle:3]: This is unrelated to the PR.

**Comment by wb9688** [comment]
I rebased and removed the unrelated `extractor/build.gradle` change.
